src/opal.c

```diff
@@ -1508,12 +1508,33 @@ make_parentheses_expression(void)
 node_s*
 make_leaf(lexeme_type_e type, lexeme_s* curr_lexeme)
 {
-  /// Create the leaf node to return
-  node_s* node = NULL;
+    /// Create the leaf node to return
+    node_s* node = NULL;
 
-  /// TODO: Replace stub implementation
+    /// If lexeme type is a string or an identifier
+    if ((type == lx_String) || (type == lx_Ident))
+    {
+        node = (node_s*) calloc (strlen(curr_lexeme->char_val) + 1, sizeof(char));
```

Comment:
You are allocating a block one more than the size of the char_val of the lexeme and casting it as a node struct. This may compile but will crash on runtime. You need to allocate & zero out 1 node struct worth of memory block and then copy string value into it's char_val location. 
```
node = calloc (1, sizeof(node_s));
node->char_val = strdup(curr_lexeme->char_val);
```